pkg/syncer/k8scloudoperator/k8scloudoperator.go

```diff
@@ -312,6 +312,16 @@ func (k8sCloudOperator *k8sCloudOperator) PlacePersistenceVolumeClaim(ctx contex
 	}
 
 	spTypes, present := sc.Annotations[spTypeAnnotationKey]
+	//if affinity rule is specified then storage class should be either vsan-sna or vsand type
```

Comment:
This is not a correct statement. A node affinity annotation is applied on a PVC when placing it on vSAN datastore, but affinitized to one of the hosts. So a vSAN policy (or StorageClass) will be used for this placement which will have a `vsan` type.

Reply:
After offline discussion with Nitish, we found that this check is correct. If the node affinity annotation is present then the storage class should be of type `vsan-sna`. For any other storage class type we can throw an error from here. For a `vsan-sna` type StorageClass, the placement engine will convert that into the corresponding StoragePool annotation on the PVC.